File: TrainWebUI/src/types/ticket.ts
import { SeatDto } from './seat';

export enum TicketStatus {
  Active = 0,
  Used = 1,
  Cancelled = 2
}

export interface TicketTypeDto {
  id?: string;
  name?: string | null;
  discountPercent?: number;
}

export interface TicketEntity {
  id?: string; // Optional when creating, required in response
  bookingId?: string; // Optional vì response từ backend không có
  seat?: SeatDto; // Backend trả về nested seat object
  ticketType?: TicketTypeDto; // Backend trả về nested ticketType
  qrCode?: string;
  status: TicketStatus | string; // Backend trả về string "Active" thay vì số
}

export const TICKET_STATUS_LABELS: Record<TicketStatus, string> = {
  [TicketStatus.Active]: 'Đang hoạt động',
  [TicketStatus.Used]: 'Đã sử dụng',
  [TicketStatus.Cancelled]: 'Đã hủy'
};

// Helper to convert string status to enum
export function parseTicketStatus(status: string | TicketStatus): TicketStatus {
  if (typeof status === 'number') return status;
  
  switch (status.toLowerCase()) {
    case 'active': return TicketStatus.Active;
    case 'used': return TicketStatus.Used;
    case 'cancelled': return TicketStatus.Cancelled;
    default: return TicketStatus.Active;
  }
}

File: TrainWeb.Domain/Domain/Ticket.cs
using TrainWeb.Domain.Enum;

namespace TrainWeb.Domain.Domain
{
    public class Ticket
    {
        public string? Id { get; }
        public string BookingId { get; }
        public string TicketNumber { get; }
        public string QrCode { get; }
        public TicketStatus Status { get; set; } = TicketStatus.Active;
        public DateTime CreatedAt { get; }
        public DateTime? ActiveAt { get; }

        public Ticket(
            string? id,
            string bookingId,
            string ticketNumber,
            string qrCode,
            TicketStatus status,
            DateTime createdAt,
            DateTime? activeAt
        )
        {
            Id = id;
            BookingId = bookingId;
            TicketNumber = ticketNumber;
            QrCode = qrCode;
            Status = status;
            CreatedAt = createdAt;
            ActiveAt = activeAt;
        }
    }
}
